src/userobjects/NEMLMaterialPropertyReset.C

```diff
@@ -0,0 +1,76 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "NEMLMaterialPropertyReset.h"
+
+registerMooseObject("BlackBearApp", NEMLMaterialPropertyReset);
+
+InputParameters
+NEMLMaterialPropertyReset::validParams()
+{
+  InputParameters params = ElementUserObject::validParams();
+
+  params.addCoupledVar("temperature", 0.0, "Temperature field");
+  params.addParam<Real>("critical_temperature", 1000.0, "Annealing temperature");
```

Comment:
Is there any reason we'd want to make this more general, to work with other variables, or maybe have more options for the condition when properties are reset? Since this class has a fairly general name, it seems like it should at least have the parameter names less specifically tied to temperature. Maybe rename "temperature" to "variable", and "critical_temperature" to "critical_value"?